packages/test/test-drivers/src/odspTestDriver.ts

```diff
@@ -38,14 +32,27 @@ export interface IOdspTestLoginInfo {
 
 type TokenConfig = IOdspTestLoginInfo & IClientConfig;
 
-interface IOdspTestDriverConfig extends TokenConfig{
+interface IOdspTestDriverConfig extends TokenConfig {
     directory: string;
     driveId: string;
 }
 
 export class OdspTestDriver implements ITestDriver {
+    // Share the tokens and driverId across multiple instance of the test driver
+    private static readonly odspTokenManager = new OdspTokenManager(odspTokensCache);
+    private static readonly driverIdPCache = new Map<string, Promise<string>>();
```

Comment:
Why you no use PromiseCache? 🥺

Reply:
e.g. if `getDriveId` below rejects, that cache key will be burned, since you don't remove it in a `.catch`, but PromiseCache takes care of that for ya.

Reply:
Created issue #5652